### ISSUE TITLE
Missing an isolated namespace [RFC]

### ISSUE THREAD

**Issue by bensteers** [rochacbruno/dynaconf#189]
**Is your feature request related to a problem? Please describe.**

The Settings object should have a way of accessing and traversing the keys loaded from the settings files without having all of the Dynaconf options in there. When I call `config.keys()` I'm getting `['SETTINGS_MODULE', 'PROJECT_ROOT_FOR_DYNACONF', 'SETTINGS_MODULE_FOR_DYNACONF', ...`

**Describe the solution you'd like**

Ideally when you call `keys()` and similar methods, it should only return values in the namespaces loaded by the file. Even if it involved doing this `config.current_env.keys()` it would be fine.

Also, having methods of exporting the current (possibly modified) settings would be nice. Basically I would like to be able to do this:

```python 
with open(output_file, 'w') as f:
    yaml.dump(config.environments['development'].as_dict(), f)
```

And have that write out a flattened version of the development namespace.

**Describe alternatives you've considered**

**Additional context**

**Comment by rochacbruno** [comment]
The same thing has been implemented in #188 for `list` command, the same solution (filtering out any key that are present in `dynaconf.default_settings`)

We should add a new method `as_dict` on settings so we keep `.keys()` working in the way it is now without breaking compatibility.

Thanks for suggestion.

**Comment by rochacbruno** [comment]
Added on #191

## Export settings as a Python dictionary

After exporting the settings to a python dictionary it is easy to use it to serialize as a JSON, YAML or any other format you may need.

### Programmatically

```py
from dynaconf import settings
settings.as_dict()  # a dict with only user defined values in current env
settings.as_dict(env='production')  # a dict with only user defined values in production env
settings.as_dict(internal=True)  # a dict with all values, user defined and dynaconf internal
```

### CLI

from your project root folder (generally the same place where you have `.env` or from where you call your scripts.

```bash
dynaconf list -o path/to/file.json
dynaconf list -e production -o path/to/file.json
```
